client/sites-dashboard/components/sites-site-coming-soon.tsx

```diff
@@ -0,0 +1,79 @@
+import styled from '@emotion/styled';
+import { __ } from '@wordpress/i18n';
+import { CSSProperties } from 'react';
+
+type Props = {
+	siteName?: string;
+	className?: string;
+	style?: CSSProperties;
+	lang?: string;
+};
+
+const Root = styled.div( {
+	display: 'flex',
+	alignItems: 'center',
+	backgroundColor: '#117ac9',
+	borderRadius: 4,
+	boxSizing: 'border-box',
+} );
+
+const comingSoonTranslations: Record< string, string > = {
+	ar: 'قريبًا',
+	de: 'Demnächst verfügbar',
+	es: 'Próximamente',
+	fr: 'Bientôt disponible',
+	he: 'בקרוב',
+	id: 'Segera hadir',
+	it: 'Presto disponibile',
+	ja: '近日公開',
+	ko: '공개 예정',
+	nb: 'Kommer snart',
+	nl: 'Komt binnenkort',
+	pt: 'Em breve',
+	ro: 'În curând',
+	ru: 'Скоро будет',
+	skr: 'جلدی آندا پئے',
+	sv: 'Kommer snart',
+	tr: 'Pek yakında',
+	zh: '即将推出',
+};
+
+const getTranslation = ( lang?: string ) => {
+	let text = __( 'Coming Soon' );
+	if ( lang ) {
+		lang = lang.split( '-' )[ 0 ];
+		if ( comingSoonTranslations[ lang ] ) {
+			text = comingSoonTranslations[ lang ];
+		}
+	}
+	return text;
+};
+
+export const SiteComingSoon = ( { siteName = '', lang = '', style }: Props ) => {
```

Comment:
It looks the approach you've taken is to propagate the `undefined`-ness of `lang` all the way down the chain. That's all good, I think that's a good approach. But to be consistent, I think it makes sense not to use a default value for `lang` here.

```suggestion
export const SiteComingSoon = ( { siteName = '', lang, style }: Props ) => {
```

Reply:
I don't think it makes much difference but I'll accept it. I also wondered why `site.lang` can be undefined in the first place as I don't think that's possible to do via the UI.